specs/hotel_spec.rb

```diff
@@ -0,0 +1,247 @@
+      @test_ob.rooms.must_be_instance_of Array
+    end
+
+    it "@rooms holds Integers 1 to 20" do
+      @test_ob.rooms[0].must_equal 1
+      @test_ob.rooms[19].must_equal 20
+    end
+
+    it "Has an instance variable @all_single_reservations that holds objects (instances of Reservation class) in an Array" do
+      @test_ob.all_single_reservations.must_be_instance_of Array
+
+      @test_ob.make_single_reservation(@room, @check_in, @check_out)
+      @test_ob.all_single_reservations[0].must_be_instance_of BookingSystem::Reservation
+    end
+
+    it "Has an instance variable @all_block_reservations that holds objects (instances of BlockReservation class) in an Array" do
+      @test_ob.block_off_a_block(3, @check_in, @check_out)
+      @test_ob.block_reservation(1, @check_in, @check_out)
+      @test_ob.all_block_reservations[0].must_be_instance_of BookingSystem::BlockReservation
+      @test_ob.all_block_reservations.must_be_instance_of Array
+    end
+
+    it "Has an instance variable @all_blocks that holds objects (instances of Block class) in an Array" do
+      @test_ob.all_blocks.must_be_instance_of Array
+
+      @test_ob.block_off_a_block(3, @check_in, @check_out)
+      @test_ob.all_blocks[0].must_be_instance_of BookingSystem::Block
+    end
+  end#initialize
+
+  describe "make_single_reservation method" do
+    it "Can be called" do
+      @test_ob.must_respond_to :make_single_reservation
+    end
+
+    it "Returns an instance of Reservation class when room is available" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out).must_be_instance_of BookingSystem::Reservation
+    end
+
+    it "Increases @all_single_reservations by 1 for each new reservation" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out)
+      @test_ob.all_single_reservations.length.must_equal 1
+    end
+
+    it "Can accept a price argument" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out, 100)
+      @test_ob.all_single_reservations[0].price.must_equal 100
+    end
+
+    it "Will assign price to DEFAULT_ROOM_PRICE 200, if given no price argument" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out)
+      @test_ob.all_single_reservations[0].price.must_equal 200
+    end
+
+    it "Raises an UnavailableRoomError if room is unavailable" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out)
+
+      proc { @test_ob.make_single_reservation(@room, @check_in, @check_out) }.must_raise BookingSystem::Hotel::UnavailableRoomError
+
+      proc { @test_ob.make_single_reservation(@room, Date.new(2017,9,5), Date.new(2017,9,10)) }.must_raise BookingSystem::Hotel::UnavailableRoomError
+    end
+
+    it "Will book two reservations with check out and check in being on the same day" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out)
+      @test_ob.make_single_reservation(@room, Date.new(2017,9,5), Date.new(2017,9,9))
```

Comment:
This is a good start to covering edge cases here, but there are many more that I'd be interested in seeing around whether or not a room is available. My full list includes:
- Not available:
  - Same dates
  - Overlaps in the front
  - Overlaps in the back
  - Completely contained
  - Completely containing
- Available:
  - Completely before
  - Completely after
  - Ends on the checkin date
  - Starts on the checkout date

These might belong in the tests for `room_available?` instead of here.